### DISCUSSION TITLE
Per-column styles

### DISCUSSION THREAD

**Discussion by tyteen4a03** [TanStack/table#2632] [General]
How can I specify in my column definition the styles I want to apply for the whole column?

**Comment by carlbrn** [comment]
You can see in this example how to add a classname or a style to the header and the cell:
[codesandbox example](https://codesandbox.io/s/github/tannerlinsley/react-table/tree/master/examples/data-driven-classes-and-styles)

You can put a condition like below.

```javascript
<th {...column.getHeaderProps([{
  className:  cell.column.Header === 'My Header' ? 'myClassName' : '',
  style: cell.column.Header === 'My Header' ? myStyleObject : {},
}])}>
```
